RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -0,0 +1,1038 @@
+/** 
+ * \file CSCSegAlgRU.cc 
+ * 
+ *  \authors V.Palichik & N.Voytishin   
+ *  \some functions and structure taken from SK algo by M.Sani and SegFit class by T.Cox 
+ */ 
+
+#include "CSCSegAlgoRU.h" 
+#include "CSCSegFit.h"
+#include "Geometry/CSCGeometry/interface/CSCLayer.h" 
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h" 
+#include "FWCore/ParameterSet/interface/ParameterSet.h" 
+#include "FWCore/MessageLogger/interface/MessageLogger.h" 
+
+#include <algorithm> 
+#include <cmath> 
+#include <iostream> 
+#include <string> 
+
+CSCSegAlgoRU::CSCSegAlgoRU(const edm::ParameterSet& ps)
+  : CSCSegmentAlgorithm(ps), myName("CSCSegAlgoRU"), sfit_(0) { 
+
+  doCollisions = ps.getParameter<bool>("doCollisions"); 
+  chi2_str_   = ps.getParameter<double>("chi2_str_"); 

```

Comment:
My comment was only about the name of the configuration parameter "chi2_str_".
The c++ double/float representation of it can have a different name.